docs/data/joy/components/autocomplete/autocomplete.md

```diff
@@ -0,0 +1,355 @@
+
+{{"demo": "GitHubLabel.js"}}
+
+### Virtualization
+
+Search within 10,000 randomly generated options. The list is virtualized thanks to [react-window](https://github.com/bvaughn/react-window).
+
+{{"demo": "Virtualize.js"}}
+
+## Custom filter
+
+The component exposes a factory to create a filter method that can be provided to the `filterOptions` prop.
+You can use it to change the default option filter behavior.
+
+```js
+import { createFilterOptions } from '@mui/material/Autocomplete';
+```
+
+### `createFilterOptions(config) => filterOptions`
+
+#### Arguments
+
+1. `config` (_object_ [optional]):
+
+- `config.ignoreAccents` (_bool_ [optional]): Defaults to `true`. Remove diacritics.
+- `config.ignoreCase` (_bool_ [optional]): Defaults to `true`. Lowercase everything.
+- `config.limit` (_number_ [optional]): Default to null. Limit the number of suggested options to be shown. For example, if `config.limit` is `100`, only the first `100` matching options are shown. It can be useful if a lot of options match and virtualization wasn't set up.
+- `config.matchFrom` (_'any' | 'start'_ [optional]): Defaults to `'any'`.
+- `config.stringify` (_func_ [optional]): Controls how an option is converted into a string so that it can be matched against the input text fragment.
+- `config.trim` (_bool_ [optional]): Defaults to `false`. Remove trailing spaces.
+
+#### Returns
+
+`filterOptions`: the returned filter method can be provided directly to the `filterOptions` prop of the `Autocomplete` component, or the parameter of the same name for the hook.
+
+In the following demo, the options need to start with the query prefix:
+
+```jsx
+const filterOptions = createFilterOptions({
+  matchFrom: 'start',
+  stringify: (option) => option.title,
+});
+
+<Autocomplete filterOptions={filterOptions} />;
+```
+
+{{"demo": "Filter.js", "defaultCodeOpen": false}}
+
+### Advanced
+
+For richer filtering mechanisms, like fuzzy matching, it's recommended to look at [match-sorter](https://github.com/kentcdodds/match-sorter). For instance:
+
+```jsx
+import { matchSorter } from 'match-sorter';
+
+const filterOptions = (options, { inputValue }) => matchSorter(options, inputValue);
+
+<Autocomplete filterOptions={filterOptions} />;
+```
+
+## Events
+
+If you would like to prevent the default key handler behavior, you can set the event's `defaultMuiPrevented` property to `true`:
+
+```jsx
+<Autocomplete
+  onKeyDown={(event) => {
+    if (event.key === 'Enter') {
+      // Prevent's default 'Enter' behavior.
+      event.defaultMuiPrevented = true;
+      // your handler code
+    }
+  }}
+/>
+```
+
+## Limitations
+
+### autocomplete/autofill
+
+Browsers have heuristics to help the user fill in form inputs.
+However, this can harm the UX of the component.
```

Comment:
Uhm... I guess this should be a bit more elaborated. Which heuristics are we talking about specifically and how it hurts?

Reply:
Agree, it does not bring much value so I have removed it.